apple/internal/ios_rules.bzl

```diff
@@ -2881,6 +3002,44 @@ use only extension-safe APIs.
     ],
 )
 
+ios_dylib = rule_factory.create_apple_rule(
+    doc = "Builds a iOS Dylib binary.",
```

Comment:
Could we add more info here, primarily what is a good use case for this rule, etc. This isnt intended to be used in `deps` of other targets right?